pkg/pubsub/adapter/converters/pubsub.go

```diff
@@ -50,7 +50,10 @@ func convertPubsub(ctx context.Context, msg *cepubsub.Message, sendMode ModeType
 	if msg.Attributes != nil && len(msg.Attributes) > 0 {
 		for k, v := range msg.Attributes {
 			// V1 attributes MUST consist of lower-case letters ('a' to 'z') or digits ('0' to '9').
-			event.SetExtension(strings.ToLower(k), v)
+			// Only setting them if the attributes are valid alphanumeric, otherwise the sdk will panic.
+			if IsAlphaNumericLowercaseLetters(k) {
```

Comment:
Is this an issue with the cloudevents sdk or the pubsub sdk?  Perhaps an error should be returned rather than silently dropping event metadata.

Reply:
I think is not an issue in neither. In v1, extensions should be lower-case letters or digits. 
It was rather our opinionated implementation to promote all pubsub attributes to extensions. If someone upstream sends an attribute that cannot be an v1 extension, do you think we should fail? I was thinking more of not to, and add as much metadata as possible, if it's valid.

Reply:
@nachocano and I talked about this. While this seems like it should be transport-specific, it's actually [a MUST in the base CloudEvents v1 specification](https://github.com/cloudevents/spec/blob/master/spec.md#attribute-naming-convention), so it's not even possible for a conformant transport to allow non-base36 characters even if the underlying technology allows it. This seems like an issue in the specification.

@nachocano please clarify in the comment that CloudEvents v1.0 requires this behavior, then I'll lgtm.

Reply:
Done, thanks @grantr !!!